joomla/joomla-cms
# [4.0 dev] Correct the layout of the modals in "Manage Folders" and "New File" in Template manager.

Pull Request for Issue #31033 and #30954.

### Summary of Changes
Modified the modal settings for both the "Manage Folders" And "New File" modal windows that appear when you edit a template.
- Modal size changed from 50vh and 60vw to 70vh and 80vw
- Layout of modal columns changed from col-md-6 & col-md-6 to col-md-4 and col-md-8 to closer match Templates: Customisation screen ratios (it uses 3/9)
- Class for .modal-body .tree-holder - sets a max-height for the tree-holder in the modal view. The .tree-holder class is also applied in the customisation screen view, and doesn't need to have it's height restricted there.
- Several responsive styles have been added to reformat the modal contents on max-width:768px

### Testing Instructions
- Familiarise yourself with the BEFORE status.
- Apply the patch. _Note that you can manually add the new class to \media\com_templates\css\admin-templates-default.css rather than needing to install the patch using NPM as the CSS file is precompiled._
- Go to: System > Site Templates > Edit the template by clicking on "Cassiopeia Details and Files"
- Click on the "New File" and "Manage Folders" buttons.
- Check what the layout of the modal looks like.
- Go to: System > Adminstrator Templates > Edit the template by clicking on "Atum Details and Files"
- Click on the "New File" and "Manage Folders" buttons.
- Check what the layout of the modal looks like.

### Actual result BEFORE applying this Pull Request
Per the various screen grabs in #31033, you before applying the PR you will have issues on screens under 1400px with the layout being cramped, and on all screens the folder tree when displayed in the modal, the tree extends over the bottom of the modal, meaning the close and/or delete buttons (depending on the screen) are disabled due to the z-index of the folder tree.
![image](https://user-images.githubusercontent.com/5515866/95651677-b86a1980-0b37-11eb-8e4f-32b408d6ed32.png)

### Expected result AFTER applying this Pull Request
_Note that the CSS change is required to make the folder tree height correct._
- Layout will change so the columns are 4:8 ratio.
- With the folder tree height capped proportional to the new height of the modal, the buttons now work.
![image](https://user-images.githubusercontent.com/5515866/95655647-8dda8980-0b54-11eb-9527-001cbbc5c270.png)

- Check also responsive layout on max-width:768px and below. 
- Uncommon as it would be for someone to modify the template via mobile, the layout now at least provides the ability to do so if needed.
![31037 Responsive](https://user-images.githubusercontent.com/5515866/95696288-3b00ef00-0c86-11eb-82d7-f623dff2ce7e.gif)
### Documentation Changes Required
No documentation changes are needed.
